src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,158 @@
+    void register_userIsNull_notOk() {
+        assertThrows(RegistrationValidationException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_duplicateLogin_notOk() {
+        user.setLogin(USER_LOGIN);
+        user.setPassword(USER_PASSWORD);
+        user.setAge(25);
+        storageDao.add(user);
+        User newUser = new User();
+        newUser.setLogin(USER_LOGIN);
+        newUser.setPassword(USER_PASSWORD);
+        newUser.setAge(25);
+        RegistrationValidationException exception
+                = assertThrows(RegistrationValidationException.class, () -> {
+                    registrationService.register(newUser);
+                });
+        assertEquals("User with this login already exists", exception.getMessage());
+    }
+
+    @Test
+    void register_invalidLogin_notOk() {
+        user.setLogin("logi");
+        user.setPassword("password2024");
+        user.setAge(32);
+        RegistrationValidationException exception
+                = assertThrows(RegistrationValidationException.class, () -> {
+                    registrationService.register(user);
+                });
+        assertEquals("Login must be at least 6 characters long", exception.getMessage());
+    }
+
+    @Test
+    void register_invalidPassword_notOk() {
+        user.setLogin("validLogin");
+        user.setPassword("pass");
+        user.setAge(22);
+        RegistrationValidationException exception
+                = assertThrows(RegistrationValidationException.class, () -> {
+                    registrationService.register(user);
+                });
+        assertEquals("Password must be at least 6 characters long", exception.getMessage());
+    }
+
+    @Test
+    void register_underageUser_notOk() {
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(17);
+        RegistrationValidationException exception
+                = assertThrows(RegistrationValidationException.class, () -> {
+                    registrationService.register(user);
+                });
+        assertEquals("Age must be at least 18 years old", exception.getMessage());
+    }
+
+    @Test
+    void register_validUser_ok() {
+        user.setLogin("validLogin1");
+        user.setPassword("validPassword1");
+        user.setAge(21);
+        User registeredUser = registrationService.register(user);
+        assertEquals(user, registeredUser);
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_LoginIsEmpty_notOk() {
+        user.setLogin("");
+        user.setPassword("password2012");
+        user.setAge(30);
+        RegistrationValidationException exception
+                = assertThrows(RegistrationValidationException.class, () -> {
+                    registrationService.register(user);
+                });
+        assertEquals("Login must be at least 6 characters long", exception.getMessage());
```

Comment:
```suggestion
        assertEquals("Login is too short", exception.getMessage());
```
Don't hardcode limits, you may forget to change them when requirements will change one day